fenix/app/src/main/java/org/mozilla/fenix/debugsettings/ui/FenixOverlay.kt

```diff
@@ -0,0 +1,45 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.debugsettings.ui
+
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.remember
+import mozilla.components.lib.state.ext.observeAsState
+import org.mozilla.fenix.compose.annotation.LightDarkPreview
+import org.mozilla.fenix.debugsettings.store.DebugDrawerAction
+import org.mozilla.fenix.debugsettings.store.DebugDrawerStore
+import org.mozilla.fenix.debugsettings.store.DrawerStatus
+import org.mozilla.fenix.theme.FirefoxTheme
+import org.mozilla.fenix.theme.Theme
+
+/**
+ * Overlay for presenting Fenix-wide debugging content.
+ */
+@Composable
+fun FenixOverlay() {
+    val debugDrawerStore = remember { DebugDrawerStore() }
+    val drawerStatus by debugDrawerStore.observeAsState(initialValue = DrawerStatus.Closed) { state ->
+        state.drawerStatus
+    }
+
+    FirefoxTheme(theme = Theme.getTheme(allowPrivateTheme = false)) {
+        DebugOverlay(
+            drawerStatus = drawerStatus,
+            onDrawerOpen = {
+                debugDrawerStore.dispatch(DebugDrawerAction.DrawerOpened)
+            },
+            onDrawerClose = {
+                debugDrawerStore.dispatch(DebugDrawerAction.DrawerClosed)
+            },
+        )
+    }
+}
+
+@LightDarkPreview
+@Composable
+private fun FenixOverlayPreview() {
+    FenixOverlay()
```

Comment:
An additional positive side effect from creating `FenixOverlay` is now the entire feature will be interactable within the previews, meaning Tab Tools and everything else we eventually plug in here will be fully functional!